core/src/autogluon/core/scheduler/hyperband.py

```diff
@@ -673,26 +685,39 @@ def load_state_dict(self, state_dict):
             logger.info('Loading Terminator State {}'.format(self.terminator))
 
     def _snapshot_tasks(self, bracket_id):
-        return {
-            k: {'config': v['config'],
-                'time': v['time_stamp'],
-                'level': 0 if v['reported_result'] is None
-                else v['reported_result'][self._time_attr]}
-            for k, v in self._running_tasks.items()
-            if v['bracket'] == bracket_id}
+        # If all brackets share a single rung level system, then all
+        # running jobs have to be taken into account, otherwise only
+        # those jobs running in the same bracket
+        all_running = not self.terminator._rung_system_per_bracket
+        tasks = dict()
+        for k, v in self._running_tasks.items():
+            if all_running or (v['bracket'] == bracket_id):
+                reported_result = v['reported_result']
+                level = 0 if reported_result is None \
+                    else reported_result[self._time_attr]
+                # It is possible to have tasks in _running_tasks which have
+                # reached self.max_t. These must not end up in the snapshot
+                if level < self.max_t:
+                    tasks[k] = {
+                        'config': v['config'],
+                        'time': v['time_stamp'],
+                        'level': level}
+        return tasks
```

Comment:
I'm not sure I understand the full flow, just wanted to get some clarity how the last block triggered after the job is done in the last bracket. Specifically, we're interested not losing the last values.

Reply:
The results for an experiment are collected in training_history, as is done for all schedulers, and there all reported results are always collected.

Snapshots are a pretty specific feature only needed for specific searchers. These searchers need to know which tasks are currently still running and will report at least one further value. This is why the tasks which have reached max_t already, are filtered out here.